snippet-extractor/src/python_child.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright 2023 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use std::{
+    io::{self, BufReader, Write},
+    path::Path,
+    process::{Child, ChildStdin, ChildStdout, Command, Stdio},
+    thread,
+    time::{Duration, Instant},
+};
+
+use rmp_serde::{
+    config::{DefaultConfig, StructMapConfig},
+    decode::ReadReader,
+};
+use serde::{de::DeserializeOwned, Serialize};
+
+pub(crate) struct PythonChild {
+    child: Child,
+    // Hint: We always write the whole package at once, so no point in using a BufWriter
+    write_to: Option<rmp_serde::Serializer<ChildStdin, StructMapConfig<DefaultConfig>>>,
+    read_from: Option<rmp_serde::Deserializer<ReadReader<BufReader<ChildStdout>>, DefaultConfig>>,
+}
+
+impl PythonChild {
+    pub(crate) fn spawn(
+        workspace: impl AsRef<Path>,
+        python_file: impl AsRef<Path>,
+        use_pipenv: bool,
+    ) -> Result<Self, io::Error> {
+        let mut cmd = if use_pipenv {
+            let mut cmd = Command::new("pipenv");
```

Comment:
Why not always use the env?

Reply:
it's using the env

Reply:
Through if you want we can change it to use `pipenv install --deploy` instead and then always use `pipenv run`.

Reply:
I did change it that way.